# Import library



In [2]:
# import library
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats

# Read data

In [3]:
df = pd.read_csv("C:/Mini Proyek/AB Testing campaign/Dataset/Ab_testing (1).csv")
df.head()

period cohort_date variant    user_id country platform traffic_source  \
0  during  2025-07-05       B  APP201334      ID  android        organic   
1  during  2025-07-04       A  APP201596      ID  android        organic   
2  during  2025-07-06       A  APP200545      TH  android        organic   
3  during  2025-07-07       B  APP203155      MY  android        organic   
4  during  2025-07-04       A  APP202354      ID  android           paid   

  device_model  age  sessions_7d  time_to_first_action_min  trial_start  \
0      low_end   54            4                       6.1            0   
1      low_end   24            2                       6.5            0   
2    mid_range   41            4                       7.3            0   
3    mid_range   40            1                       2.3            0   
4    mid_range   49            6                       2.4            1   

   paid_subscriber  revenue_30d  
0                0   444.616527  
1                0   424.835708  
2                0   393.086785  
3                0   502.190971  
4                0   432.384427

# Data Preparation

In [4]:
aov_A = df[(df['variant']=='A') & (df['trial_start'] == 1) & (df['period'] == 'during')]
aov_B = df[(df['variant']=='B') & (df['trial_start'] == 1) & (df['period'] == 'during')]

In [5]:
aov_B.shape[0]

42

# Cek Normality

In [6]:
shapiro_A = stats.shapiro(aov_A['revenue_30d'])
shapiro_B = stats.shapiro(aov_B['revenue_30d'])

print("If p-value < 0.05 tidak terdistribusi normal\n")

print("Shapiro-Wilk test for Variant A: p-value={:.2f}".format(shapiro_A.pvalue))
print("Shapiro-Wilk test for Variant B: p-value={:.2f}".format(shapiro_B.pvalue))

If p-value < 0.05 tidak terdistribusi normal

Shapiro-Wilk test for Variant A: p-value=0.06
Shapiro-Wilk test for Variant B: p-value=0.49


# Cek Variasi

In [7]:
cek_variansi = stats.levene(aov_A['revenue_30d'], aov_B['revenue_30d'])

print("If p-value < 0.05 tidak homogen (equal variance)")

print("Nilai p-value: {:.2f}".format(cek_variansi.pvalue))

If p-value < 0.05 tidak homogen (equal variance)
Nilai p-value: 0.92


# Uji Statistik
- H0: varian A dan varian B tidak ada bedanya terhadap order value.
- H1: Varian B mampu meningkatkan order value dibanding varian A

In [8]:
whitney = stats.mannwhitneyu(aov_A['revenue_30d'], aov_B['revenue_30d'])

if whitney.pvalue < 0.05:
  print("Reject H0")
else:
  print("Accept H0")

print("Nilai p-value: {:.2f}".format(whitney.pvalue))

Reject H0
Nilai p-value: 0.00


# Uji Kenaikan Bisnis

In [9]:
mean_A = aov_A['revenue_30d'].mean()
mean_B = aov_B['revenue_30d'].mean()

diff = mean_B - mean_A

print('Rata-rata variant A: {:.2f} dan variant B: {:.2f}'.format(mean_A, mean_B))
print("Perbedaan antara kedua variant: {:.2f}".format(diff))

Rata-rata variant A: 406.05 dan variant B: 486.88
Perbedaan antara kedua variant: 80.83


In [10]:
mean_A = aov_A['revenue_30d'].mean()
mean_B = aov_B['revenue_30d'].mean()

uplift_abs = (mean_B - mean_A)
uplift_perc = round((mean_B - mean_A)/mean_A*100,2)

print("Uplift Absolut; ", uplift_abs)
print("Uplift Percent; {}%".format(uplift_perc))

Uplift Absolut;  80.83226887231848
Uplift Percent; 19.91%


Key insight : variant B menghasilkan mean revenue 80.83 lebih tinggi dari A, atau +19.91% lebih besar.

In [11]:
median_A = aov_A['revenue_30d'].median()
median_B = aov_B['revenue_30d'].median()

diff = median_B - median_A

print('Rata-rata variant A: {:.2f} dan variant B: {:.2f}'.format(median_A, median_B))
print("Perbedaan antara kedua variant: {:.2f}".format(diff))

Rata-rata variant A: 404.47 dan variant B: 480.78
Perbedaan antara kedua variant: 76.31


In [12]:
median_A = aov_A['revenue_30d'].median()
median_B = aov_B['revenue_30d'].median()

uplift_abs = (median_B - median_A)
uplift_perc = round((median_B - median_A)/median_A*100,2)

print("Uplift Absolut; ", uplift_abs)
print("Uplift Percent; {}%".format(uplift_perc))

Uplift Absolut;  76.31072644696678
Uplift Percent; 18.87%


Key insight : variant B menghasilkan median revenue 76.31 lebih tinggi dari A, atau +18.87% lebih besar.